### PR TITLE
4.1.0_dev

### PR DESCRIPTION
**Pull request by mirr254**
## What does this pr do?  
Updates to the new Gluu-gateway 4.1

## How should this be manually tested

1. Create a common docker network 
 `docker network create kong-net`

 2. Start your database
 `docker run -d --name kong-database --network kong-net -p 5432:5432 -e "POSTGRES_USER=kong" -e "POSTGRES_DB=kong" postgres:9.6`

3. Prepare your database.
    `docker run --rm \
     --network=kong-net \
     -e "KONG_DATABASE=postgres" \
     -e "KONG_PG_HOST=kong-database" \
     -e "KONG_CASSANDRA_CONTACT_POINTS=kong-database" \
     gluufederation/gluu-gateway:4.1.0_dev kong migrations bootstrap`

4. Start gg-api
     ` docker run -d --name kong \
     --network=kong-net \
     -e "KONG_DATABASE=postgres" \
     -e "KONG_PG_HOST=kong-database" \
     -e "KONG_CASSANDRA_CONTACT_POINTS=kong-database" \
     -e "KONG_PROXY_ACCESS_LOG=/dev/stdout" \
     -e "KONG_ADMIN_ACCESS_LOG=/dev/stdout" \
     -e "KONG_PROXY_ERROR_LOG=/dev/stderr" \
     -e "KONG_ADMIN_ERROR_LOG=/dev/stderr" \
     -e "KONG_ADMIN_LISTEN=0.0.0.0:8001, 0.0.0.0:8444 ssl" \
     -p 8000:8000 \
     -p 8443:8443 \
     -p 8001:8001 \
     -p 8444:8444 \
     gluufederation/gluu-gateway:4.1.0_dev`

4. Test if gg is up, running and ready to accept connections
    `curl -i http://localhost:8001/` 
     The expected output for this command if GG is running correctly is shown in the screenshot below.

## Screenshots if available

Screenshot 2019-12-31 at 17 52 45

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[submodule "third-party/lua-resty-lrucache"]
```
Comment (iromli) [inline .gitmodules]: rename this file as `gitmodules` to avoid being initialized accidentally

File: Dockerfile [hunk 2]
```diff
@@ -1,18 +1,25 @@
 FROM kong:1.3.0-alpine
 
 RUN apk update \
-    && apk add --no-cache --virtual build-deps unzip
+    && apk add -Uuv --no-cache --virtual build-deps git openssh bash curl
```
Comment (iromli) [inline Dockerfile]: do we need `openssh`, `bash`, and `curl` here?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,11 @@
 # docker-gg
 Docker container repo for Gluu Gateway (GG)
+
+We have used the tool `ONVAULT` can be found here [ONVAULT](https://github.com/dockito/vault) to make use of ssh keys while building the image. ONVAULT makes sure your keys are deleted immediately after use in the build process.
```
Comment (iromli) [inline README.md]: you can remove ONVAULT docs from this file

**Comment by mirr254** [comment]
On it @iromli

**Comment by mirr254** [comment]
PR updated.

**Comment by iromli** [comment]
@mirr254 please point the PR to 4.1 branch.

**Comment by mirr254** [comment]
> @mirr254 please point the PR to 4.1 branch.

Done

**Pull request merged**